app/diversity/post.controller.js

```diff
@@ -0,0 +1,45 @@
+const { body } = require('express-validator')
+const { setSentencePlanComment } = require('../../common/data/sentencePlanningApi')
+const { getDiversity } = require('./get.controller')
+const { countWords } = require('../../common/utils/util')
+
+const wordsAllowed = 250
+let tooManyWords = false
+
+const validationRules = () => {
+  return [
+    body('diversity')
+      .isLength({ min: 1 })
+      .withMessage('Record how you will take account of diversity factors'),
+    body('diversity')
+      .custom(value => {
+        tooManyWords = countWords(value) > wordsAllowed
+        return !tooManyWords
+      })
+      .withMessage('Response to diversity factors must be 250 words or fewer'),
+  ]
+}
+
+const postDiversity = async (req, res) => {
+  if (req.errors) {
+    const renderInfo = {}
+    let wordsOver = false
+    if (tooManyWords || req.tooManyWords) {
+      wordsOver = countWords(req.body.diversity) - wordsAllowed
+    }
+    renderInfo.wordsOver = wordsOver
+    req.renderInfo = renderInfo
+    await getDiversity(req, res)
+  } else if (req.body.diversity) {
+    const comment = [
+      {
+        comment: req.body.diversity,
+        commentType: 'YOUR_RESPONSIVITY',
+      },
+    ]
+    await setSentencePlanComment(req.params.planid, comment, req.session['x-auth-token'])
+    res.redirect('./need-to-know')
+  }
```

Comment:
Is there a genuine case for neither of the conditions to be true, in which case do nothing?
It feels like we should be checking for a pass or a fail scenario, so instead:
```if(req.body.diversity && req.errors){ 
//do stuff
}
// else error
```

Reply:
I'd probably de-structure the req object too. 🤔

Reply:
Probably got a fair point - will take a look at this

Reply:
> I'd probably de-structure the req object too. 🤔

I need the full req object for the `await getDiversity(req, res)` line so won't do this

Reply:
Yeah, I didn't mean in the attributes. I prefer de-structured objects so I'd probably have had the second line as:
const {errors, body: { diversity }, renderInfo,  params: { planId }, session: { ('x-auth-token') } } = req

Its a style call though I guess 🤷‍♀